roles/kubernetes/client/tasks/main.yml

```diff
@@ -1,15 +1,11 @@
 ---
-- name: Set first kube master
-  set_fact:
-    first_kube_master: "{{ hostvars[groups['kube-master'][0]]['access_ip'] | default(hostvars[groups['kube-master'][0]]['ip'] | default(hostvars[groups['kube-master'][0]]['ansible_default_ipv4']['address'])) }}"
-
 - name: Set external kube-apiserver endpoint
   set_fact:
     external_apiserver_endpoint: >-
       {%- if loadbalancer_apiserver is defined and loadbalancer_apiserver.port is defined -%}
       https://{{ apiserver_loadbalancer_domain_name }}:{{ loadbalancer_apiserver.port|default(kube_apiserver_port) }}
       {%- else -%}
-      https://{{ first_kube_master }}:{{ kube_apiserver_port }}
+      https://{{ kube_apiserver_address }}:{{ kube_apiserver_port }}
```

Comment:
This value renders to localhost. This admin kubeconfig file is meant to be copied to hosts outside of the kubernetes cluster. You can't put https://localhost:6443 into that file because it will be used on non-k8s hosts.

Reply:
no， it is define in https://github.com/kubernetes-incubator/kubespray/blob/master/roles/kubespray-defaults/defaults/main.yaml
kube_apiserver_address: "{{ ip | default(ansible_default_ipv4['address']) }}"

Reply:
ok